stm32,i2c: Fix large I2C transactions on I2C V1

trustngotech opened the pull request:
Previous commit #43577 added support of large transactions on I2C v2, this commit implements some changes to also add support of large transactions on I2C v1. Some refactoring is also done to put the code in the right source files.

Fixes zephyrproject-rtos#58866

erwango commented on drivers/i2c/i2c_ll_stm32_v2.c:
```diff
@@ -738,3 +738,52 @@ int stm32_i2c_configure_timing(const struct device *dev, uint32_t clock)
+		      uint16_t periph)
+{
+	/*
+	 * Perform a I2C transaction, while taking into account the STM32 I2C
```
To make sure people understand this differenciation was made on purpose, I propose to make it clear:
```suggestion
	 * Perform a I2C transaction, while taking into account the STM32 I2C V2
```

erwango reviewed (changes requested):
Thanks for sharing this change!

I have not been able to test, some comments inline.
Also, please remove `stm32_i2c_msg_write/read` definitions from of `drivers/i2c/i2c_ll_stm32.h` and make them static in _v1 and _v2 files.

mimok committed:
stm32,i2c: Fix large I2C transactions on I2C V1

Previous commit added support of large transactions on I2C v2,
this commit implements some changes to also add support
of large transactions on I2C v1. Some refactoring is also done
to put the code in the right source files.

Fixes zephyrproject-rtos#58866

Signed-off-by: Michael Grand <m.grand@trustngo.tech>

trustngotech commented on drivers/i2c/i2c_ll_stm32_v2.c:
```diff
@@ -738,3 +738,52 @@ int stm32_i2c_configure_timing(const struct device *dev, uint32_t clock)
+
+int stm32_i2c_transaction(const struct device *dev,
+		      struct i2c_msg msg, uint8_t *next_msg_flags,
+		      uint16_t periph)
```
done

trustngotech commented:
Zephyr CI still complaining about coding style issues. Hope this time things will be ok...

trustngotech commented:
Unfortunately, CI is still complaining about either indentation (tab vs. spaces etc.) or lines being to long. Don't want to fight with CI so indentation will stay as is because if this is not blocking.
